File: sdk/python/examples/single_agent.py
"""Single agent example.

Starts one agent run and prints the kind of each event. Runs fully offline.

Usage::

    python -m examples.single_agent
"""

import asyncio
import json

import ancora


async def main() -> None:
    rt = ancora.Runtime()
    spec = ancora.AgentSpec(
        name="demo-agent",
        model_id="local-model",
        instructions="Respond to the user.",
    )
    agent = ancora.Agent(rt, spec)
    run = await agent.run()
    print(f"started run: {run.run_id}")

    async for raw in run.stream_events():
        ev = json.loads(raw)
        kind = ev["kind"]
        if kind == "token":
            print(f"  token: {ev['text']!r}")
        else:
            print(f"  event: {kind}")

    rt.free()
    print("done.")


if __name__ == "__main__":
    asyncio.run(main())

File: examples/java/single-agent/src/main/java/io/ancora/examples/SingleAgent.java
package io.ancora.examples;

import io.ancora.*;

public class SingleAgent {
    public static void main(String[] args) throws Throwable {
        try (Agent agent = new Agent()) {
            AgentSpec spec = new AgentSpec(
                "claude-3-5-haiku-20241022",
                "You are a concise assistant. Answer in one sentence.",
                null, null, null);

            RunHandle handle = agent.run(spec);
            System.out.println("Run ID: " + handle.runId());

            for (RunEvent ev : handle.events()) {
                switch (ev) {
                    case RunEvent.Started s ->
                        System.out.println("[started]");
                    case RunEvent.Token t ->
                        System.out.print(t.text());
                    case RunEvent.ToolCall tc ->
                        System.out.println("[tool_call] " + tc.name() + " <- " + tc.input());
                    case RunEvent.Completed c ->
                        System.out.println("\n[completed]");
                    default -> {}
                }
            }
        }
    }
}
